### DISCUSSION TITLE
incompatible mods but can't find newer versions of fabric or spruceUI for MidnightControls

### DISCUSSION THREAD

**Discussion by darkpitsimp** [FabricMC/fabric-api#3808] [Player support]
### Description

Incompatible mods found!
net.fabricmc.loader.impl.FormattedException: Some of your mods are incompatible with the game or each other!
A potential solution has been determined, this may resolve your problem:
	 - Install fabric, version 0.71.0 or later.
	 - Replace mod 'SpruceUI' (spruceui) 1.5.2 and mod 'SpruceUI' (spruceui) 5.0.3+1.20.4 with mod 'SpruceUI' (spruceui), version 3.2.0 or later.
More details:
	 - Mod 'MidnightControls' (midnightcontrols) 1.9.5+1.20 requires version 0.71.0 or later of fabric, which is missing!
	 - Mod 'MidnightControls' (midnightcontrols) 1.9.5+1.20 requires version 3.2.0 or later of mod 'SpruceUI' (spruceui), which can't be loaded due to other constraints!
	 - Mod 'SpruceUI' (spruceui) 1.5.2 requires any version of fabric, which is missing!
	at net.fabricmc.loader.impl.FormattedException.ofLocalized(FormattedException.java:51)
	at net.fabricmc.loader.impl.FabricLoaderImpl.load(FabricLoaderImpl.java:195)
	at net.fabricmc.loader.impl.launch.knot.Knot.init(Knot.java:146)
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:68)
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23)

I can't find fabric 0.71.0 or SpruceUI 3.2.0

**Comment by Poopooracoocoo** [comment]
Make sure all your mods are for your Minecraft version. 1.20.1 mods will not work on 1.20.6. 
By "fabric" it's referring to a mod called Fabric API.

SpruceUI is usually included in other mods. If you share a screenshot of your mods folder and tell us your Minecraft version we can help.

**Reply by darkpitsimp** [reply]
![java mod folder](https://github.com/FabricMC/fabric/assets/170989333/53e0c4cb-21c7-4edd-a234-599ddecf755e)
i have 1.20.6

**Reply by Poopooracoocoo** [reply]
You've got a Forge 1.12.2 version of Pixelmon. 
You've got a Fabric 1.20.1 version of Midnight Controls. And you've got a Fabric 1.16 version of Spruce UI. 

Remove Spruce UI. And get the correct version of Midnight Controls.

You should set up seperate folders for your Forge 1.12 profile and your Fabric 1.20.6 profile. See https://fabricmc.net/wiki/player:tutorials:install_mcl:windows#step_2_optionaledit_the_game_directory_of_the_new_fabric_installation

If you don't have a Forge 1.12 profile and wanted to only use Fabric, you can switch to Minecraft 1.20.1 and switch to Cobblemon.

**Reply by darkpitsimp** [reply]
the file i have in my mods folder for midnight controls comes from the page that says it's for 1.20.6 
https://modrinth.com/mod/midnightcontrols/version/1.9.5+1.20.5

**Reply by Poopooracoocoo** [reply]
Oh all good then, sorry. Just remove Spruce UI
